# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1327841585 0
#      Sun Jan 29 12:53:05 2012 +0000
# Node ID b4d7de953066dc312bdfea0bf610e4035261eb75
# Parent  fd8b8f0f68b93a38b414c03b65cbfe78f86b8759
Export more required symbols [Win32].

* liboctave/regexp.h (class regexp): Decorate with OCTAVE_API
* src/graphics.h.in (gtk_manager::create_instance, gtk_manager::instance):
Decorate with OCTINTERP_API.

diff --git a/liboctave/regexp.h b/liboctave/regexp.h
--- a/liboctave/regexp.h
+++ b/liboctave/regexp.h
@@ -28,17 +28,19 @@ along with Octave; see the file COPYING.
 #include <sstream>
 #include <string>
 
 #include "Array.h"
 #include "Matrix.h"
 #include "base-list.h"
 #include "str-vec.h"
 
-class regexp
+class
+OCTAVE_API
+regexp
 {
 public:
 
   class opts;
   class match_data;
 
   regexp (const std::string& pat = "",
           const regexp::opts& opt = regexp::opts (),
diff --git a/src/graphics.h.in b/src/graphics.h.in
--- a/src/graphics.h.in
+++ b/src/graphics.h.in
@@ -2305,17 +2305,17 @@ private:
 
   // FIXME -- default toolkit should be configurable.
 
   gtk_manager (void)
     : dtk ("gnuplot"), available_toolkits (), loaded_toolkits () { }
 
   ~gtk_manager (void) { }
 
-  static void create_instance (void);
+  OCTINTERP_API static void create_instance (void);
 
   static bool instance_ok (void)
   {
     bool retval = true;
 
     if (! instance)
       create_instance ();
 
@@ -2326,17 +2326,17 @@ private:
         retval = false;
       }
 
     return retval;
   }
 
   static void cleanup_instance (void) { delete instance; instance = 0; }
 
-  static gtk_manager *instance;
+  OCTINTERP_API static gtk_manager *instance;
 
   // The name of the default toolkit.
   std::string dtk;
 
   // The list of toolkits that we know about.
   std::set<std::string> available_toolkits;
 
   // The list of toolkits we have actually loaded.
